src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -518,7 +518,17 @@ private function createEncoder($config, ContainerBuilder $container)
         // Argon2i encoder
         if ('argon2i' === $config['algorithm']) {
             if (!Argon2iPasswordEncoder::isSupported()) {
-                throw new InvalidConfigurationException('Argon2i algorithm is not supported. Please install the libsodium extension or upgrade to PHP 7.2+.');
+                // Debugging on Heroku and derivatives is a lot of work, show specific error to save end users some time
+                if (false !== strpos($_ENV['PATH'], 'heroku')) {
+                    throw new InvalidConfigurationException('Argon2i algorithm is not supported on Heroku, use BCrypt instead.');
+                }
+                if (\PHP_VERSION_ID >= 70200) {
+                    // Argon2 support got added in libsodium 1.0.9
+                    if (version_compare(\SODIUM_LIBRARY_VERSION, '1.0.9', '<')) {
+                        throw new InvalidConfigurationException('Argon2i algorithm is not supported. You probably installed libsodium18 instead of libsodium23. Please update your libsodium package.');
+                    }
+                }
```

Comment:
remove the php check version. If the constant exists, and the version is wrong, then trigger the exception because the libsodium version is wrong.

(Because libsodium can be compiled against PHP < 7.2, can't remember the lowest version supported though)

Reply:
The check is to show a upgrade notice.